client/blocks/term-form-dialog/index.jsx

```diff
@@ -134,11 +135,42 @@ class TermFormDialog extends Component {
 		this.props.bumpStat( 'taxonomy_manager', statLabels.mc );
 		this.props.recordGoogleEvent( 'Taxonomy Manager', statLabels.ga );
 
-		savePromise.then( ( savedTerm ) => {
-			this.setState( { saving: false } );
-			this.props.onSuccess( savedTerm );
-			this.closeDialog();
-		} );
+		savePromise
+			.then( ( savedTerm ) => {
+				this.setState( { saving: false } );
+				this.props.onSuccess( savedTerm );
+				this.closeDialog();
+
+				const successText = isNew
+					? translate( 'Added new %(taxonomyName)s successfully.', {
+							args: { taxonomyName: taxonomy },
+					  } )
+					: translate( 'Updated %(taxonomyName)s successfully.', {
+							args: { taxonomyName: taxonomy },
+					  } );
+
+				this.props.successNotice( successText, {
+					id: 'taxonomy-manager-save',
+					duration: 4000,
+				} );
+			} )
+			.catch( () => {
+				this.setState( { saving: false } );
+				this.closeDialog();
+
+				const errorText = isNew
+					? translate( 'Failed to create new %(taxonomyName)s.', {
```

Comment:
I opted to go with a set message here because the endpoint only ever returns "Unknown taxonomy": https://github.com/Automattic/jetpack/blob/fb912fede775e2851e326b3e0991b62696badabb/projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-taxonomy-endpoint.php#L302-L308

That's not particularly helpful (and probably more confusing!) for a user who has definitely created a taxonomy.